core/unit_test/TestExecutionSpace.hpp

```diff
@@ -54,10 +54,9 @@ struct CheckClassWithExecutionSpaceAsDataMemberIsCopyable {
   Kokkos::DefaultHostExecutionSpace host;
 
   KOKKOS_FUNCTION void operator()(int, int& e) const {
-    auto copy = *this;
```

Comment:
This produced problems in the RDC build such as
```
nvlink error   : Undefined reference to '_ZN6Kokkos4Impl13HostSharedPtrINS0_12CudaInternalEEC1ERKS3_' in 'CMakeFiles/KokkosCore_UnitTest_Cuda1.dir/cuda/TestCuda_ExecutionSpace.cpp.o'

make[2]: *** [core/unit_test/CMakeFiles/KokkosCore_UnitTest_Cuda1.dir/build.make:905: core/unit_test/KokkosCore_UnitTest_Cuda1] Error 255
```

Reply:
This was the issue we saw previously?

Reply:
Yes, and I had to use two calls for MSVC in another place.